tests/slow/src/test/scala/tests/sbt/SbtServerSuite.scala

```diff
@@ -0,0 +1,138 @@
+package tests.sbt
+
+import java.util.concurrent.TimeUnit
+
+import scala.meta.internal.builds.SbtBuildTool
+import scala.meta.internal.builds.SbtDigest
+import scala.meta.internal.metals.Messages
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.ServerCommands
+import scala.meta.internal.metals.{BuildInfo => V}
+import scala.meta.io.AbsolutePath
+
+import tests.BaseImportSuite
+import tests.ScriptsAssertions
+
+/**
+ * TODO make a BaseLspSuite powered by sbt
```

Comment:
This is probably the largest follow-up that I think I'd want to work on after this. Mainly because while this suite introduces _very_ basic test for the sbt server, our entire BaseLsp suite is focused on Bloop. I'd like to either refactor this or just introduce another base suite that is focused on sbt BSP support. This would need to happen before I'd ever even consider changing from Bloop to sbt as the default for sbt builds.

Reply:
I agree this will be needed, but for sure we can skip it in this PR. It's already a lot of great work!

Reply:
We should create an issue for it though, otherwise this might be easily missed.

Reply:
Sounds good. I've created an issue here for this https://github.com/scalameta/metals/issues/2163. We can discuss a bit, see what the best way to move forward is, and then dive in.